src/app/api/quality_matrix/quality_matrix.py

```diff
@@ -272,13 +132,28 @@ async def stored_in_timeline(data: QUALITY_MATRIX_RETURN_TYPE, database: Databas
     await database.disconnect()
 
 
-async def quality_matrix() -> QUALITY_MATRIX_RETURN_TYPE:
+async def items_in_response(response: Response) -> dict:
+    return response.aggregations.to_dict().items()
+
+
+async def quality_matrix(
+    node_id: UUID = PORTAL_ROOT_ID,
+    match_keyword: str = f"{PROPERTIES}.{REPLICATION_SOURCE_ID}",
+) -> QUALITY_MATRIX_RETURN_TYPE:
     properties = get_properties()
-    output = {k: {} for k in properties}
-    for replication_source, total_count in all_sources().items():
-        response = all_missing_properties(properties, replication_source)
-        for key, value in response.aggregations.to_dict().items():
-            output[key] |= missing_fields(value, total_count, replication_source)
+    columns = all_sources()
+    mapping = {key: key for key in columns.keys()}  # identity mapping
+    return await _quality_matrix(columns, mapping, match_keyword, node_id, properties)
+
 
+async def _quality_matrix(columns, mapping, match_keyword, node_id, properties):
+    output = {k: {} for k in properties}
+    for column, total_count in columns.items():
```

Comment:
- I think the whole mapping argument is redundant: As documented above it always is an identity map holding the keys of the column dictionary. 
- Hence in line 152 we could simply loop over `columns.keys()` which should be the same set of strings.
- In line 157 `mapping[column]` will always be `column`
- Which eliminates the usage of the `mapping` argument.
- Also the if statement becomes redundant, because column will always be in `mapping.keys()`

Am I missing something there?

Reply:
This indicates, that the function in its current state is ... ambiguous? At least it is not clear for which different cases it is being used. I refactor the names to a more telling version.